File: agent/internal/tool/git_status.go
package tool

import (
	"context"
	"fmt"

	"github.com/xz-ephyr/raw-code/agent/pkg/api"
)

func gitStatusTool() ToolDef {
	return ToolDef{
		Definition: api.ToolDefinition{
			Name:        "git_status",
			Description: "Show the working tree status — modified, staged, and untracked files in the git repository.",
			Category:    "git",
			Parameters: map[string]api.ParamDef{
				"path": {Type: "string", Description: "Path to the git repository", Required: false, Default: "."},
			},
		},
		Handler: func(ctx context.Context, e *Executor, params map[string]any) (any, error) {
			path, _ := params["path"].(string)
			if path == "" {
				path = "."
			}
			out, err := gitCmd(ctx, path, "status", "--short", "--branch")
			if err != nil {
				return nil, fmt.Errorf("git status failed: %w", err)
			}
			return map[string]any{"path": path, "output": out}, nil
		},
	}
}

File: src/services/tools/git/gitStatus.ts
import { z } from 'zod';
import type { ToolDef } from '../types';

export const gitStatusTool: ToolDef = {
  name: 'git_status',
  description: 'Show the working tree status. Returns staged, unstaged, and untracked changes in the git repository.',
  category: 'git',
  inputSchema: z.object({
    repoPath: z.string().describe('Absolute path to the git repository root.'),
  }),
  execute: async ({ repoPath }) => {
    const backendUrl = import.meta.env.VITE_API_URL || 'http://localhost:3001';
    try {
      const res = await fetch(`${backendUrl}/tools`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ tool: 'git_status', params: { repoPath } }),
      });
      if (!res.ok) throw new Error(`Backend error (${res.status})`);
      return res.json();
    } catch (error: any) {
      if (error.message?.includes('fetch')) {
        throw new Error('Git tools require the Express backend to be running (http://localhost:3001) or the Tauri desktop app');
      }
      throw error;
    }
  },
};
